src/System.Diagnostics.EventLog/tests/Helpers.cs

```diff
@@ -13,22 +13,22 @@ internal class Helpers
         public static bool IsElevatedAndSupportsEventLogs { get => AdminHelpers.IsProcessElevated() && SupportsEventLogs; }
         public static bool SupportsEventLogs { get => PlatformDetection.IsNotWindowsNanoServer; }
 
-        public static void RetryOnWin7(Action func)
+        public static void RetryOnWin7(Action func, bool tryOnAllPlatforms = true)
```

Comment:
Instead of a bool I would have another method (or pair of methods) named RetryOnAllPlatforms(). 

In your RetryOnWin7 methods they would that check whether they are on Windows 7: if not, then just call func(), but if on Windows 7, then call RetryOnAllPlatforms(), where the retries happen.

That makes the callers clearer - easier to read.